packages/core/src/tools/shell.ts

```diff
@@ -101,6 +101,44 @@ export class ShellTool extends BaseTool<ShellToolParams, ToolResult> {
     return description;
   }
 
+  private _resolveDirectory(directory: string): {
+    paths: string[];
+    error: string | null;
+  } {
+    const workspaceContext = this.config.getWorkspaceContext();
+    const workspaceDirs = workspaceContext.getDirectories();
+    const possiblePaths: string[] = [];
+
+    for (const dir of workspaceDirs) {
+      const resolvedPath = path.join(dir, directory);
+      const exists = fs.existsSync(resolvedPath);
+      const within = workspaceContext.isPathWithinWorkspace(resolvedPath);
+      if (exists && within) {
+        if (!possiblePaths.includes(resolvedPath)) {
+          possiblePaths.push(resolvedPath);
+        }
+      }
+    }
+
+    if (possiblePaths.length === 0) {
+      return {
+        paths: [],
+        error: `Directory '${directory}' is not a registered workspace directory.`,
+      };
+    }
+
+    if (possiblePaths.length > 1) {
+      return {
+        paths: [],
+        error: `Directory '${directory}' is ambiguous as it exists in multiple workspace locations: ${possiblePaths.join(
+          ', ',
+        )}`,
+      };
+    }
+
+    return { paths: possiblePaths, error: null };
```

Comment:
Instead of returning an array of possible paths we should just return the final possible path because otherwise it's ambiguous OR not found 😄

Reply:
yes makes sense! Changed now.